File: exampointSpring/src/main/java/com/example/exampoint/service/QuizService.java
package com.example.exampoint.service;

import java.util.Set;

import com.example.exampoint.entity.Quiz;

public interface QuizService {
	
	public Quiz addQuiz (Quiz quiz);
	
	public Quiz updateQuiz (Quiz quiz);
	
	public Set<Quiz> getQuizzes();
	
	public Quiz getQuiz(Long quizId);	

	public void deleteQuiz(Long quizId);

}

File: exampointAngular/src/app/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private http : HttpClient) { }


  public quizzes(){
    return this.http.get(`${baseUrl}/quiz/`);
  }


  // public addQuiz(quiz){
  //   this.http.post(`${baseUrl}/quiz/`, quiz)
  // }

  addQuiz(quiz: any): Observable<any> {
    return this.http.post(`${baseUrl}/quiz/`, quiz);
  }



   deleteQuiz(qId: any){
    return this.http.delete(`${baseUrl}/quiz/${qId}`);
  }


  public getQuiz(qId: any){
    return this.http.get(`${baseUrl}/quiz/${qId}`);
  }


  public updateQuiz(quiz: any){
    return this.http.put(`${baseUrl}/quiz/`, quiz)
  }
}
